media-libs/portmidi/portmidi-2.0.3~git.recipe

```diff
@@ -0,0 +1,155 @@
+SUMMARY="Cross-Platform MIDI IO"
+DESCRIPTION="PortMidi is a cross-platform MIDI input/output library."
+HOMEPAGE="https://github.com/PortMidi/portmidi"
+COPYRIGHT="1999-2000 Ross Bencina and Phil Burk
+	2001-2009 Roger B. Dannenberg"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="7db20989f1571b27bd01cf9418361e988bdcf99a"
+SOURCE_URI="https://github.com/PortMidi/portmidi/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="96872e3c89339c9cbe44d5d86de1745ced7d803563a3ef71840aa9b120497e60"
+SOURCE_DIR="portmidi-$srcGitRev"
+ADDITIONAL_FILES="pmdefaults.rdef.in"
+
+ARCHITECTURES="all"
+SECONDARY_ARCHITECTURES="x86"
+
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	portmidi$secondaryArchSuffix = $portVersion
+	lib:libportmidi$secondaryArchSuffix = $portVersionCompat
+	"
+REQUIRES="
+	haiku${secondaryArchSuffix}
+	"
+
+PROVIDES_devel="
+	portmidi${secondaryArchSuffix}_devel = $portVersion
+	devel:libportmidi$secondaryArchSuffix = $portVersionCompat
+	"
+REQUIRES_devel="
+	portmidi$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc${secondaryArchSuffix}
+	cmd:ld${secondaryArchSuffix}
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+# the java bindings do not work with x86_gcc2 because java is not available for it
+# (it would build but fail at runtime with an "UnsatisfiedLinkError: missing library")
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	PROVIDES_devel+="
+		devel:libpmjni$secondaryArchSuffix = $portVersionCompat
+		"
+	REQUIRES_devel+="
+		portmidi${secondaryArchSuffix}_java == $portVersion
+		"
+
+	SUMMARY_java="Cross-Platform MIDI IO (Java bindings)"
+	PROVIDES_java="
+		portmidi${secondaryArchSuffix}_java = $portVersion
+		lib:libpmjni$secondaryArchSuffix = $portVersionCompat
+		"
+	REQUIRES_java="
+		haiku${secondaryArchSuffix}
+		portmidi$secondaryArchSuffix == $portVersion base
+		cmd:java
+		java:runtime
+		"
+
+	BUILD_REQUIRES+="
+		java:environment
+		"
+
+	BUILD_PREREQUIRES+="
+		cmd:javac
+		"
+fi
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+
+	extraCMakeArgs=
+	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+		source /system/data/profile.d/openjdk.sh
+		extraCMakeArgs="-DBUILD_JAVA_NATIVE_INTERFACE=ON -DBUILD_PMDEFAULTS=ON"
+	fi
+	cmake .. -DCMAKE_BUILD_TYPE=Release $cmakeDirArgs \
+		-DBUILD_PORTMIDI_TESTS=ON $extraCMakeArgs
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build
+	make install
+	cd ..
+
+	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+		cp pm_java/pmdefaults/pmdefaults.jar $libDir
```

Comment:
Is this the correct location for a jar file?

Reply:
No idea. Where are the other .jars that come with java placed?

Reply:
OpenJDK itself is intalled in $libDir/openjdk$majorVersion.

I looked at other recipes in the dev-java category. abcl uses $libDir, ant_core uses $libDir/ant, rxtx uses the lib/ext directory of openjdk8 itself (hard-coded version and only for x86 it seems).

So I think, according to these examples, $libDir is probably okay.